src/core/api.proto

```diff
@@ -94,6 +122,15 @@ message InferRequestHeader
     //@@       for tensors with a non-fixed-size datatype (like STRING).
     //@@
     uint64 batch_byte_size = 3;
+
+    //@@    .. cpp:var:: InferSharedMemory shared_memory
+    //@@
+    //@@       It is the location in shared memory that contains the tensor data
+    //@@       for this input. Using shared memory is optional but if this
+    //@@       message is used, all fields are required. The memory must be
```

Comment:
It is too confusing to try to explain contiguous vs non-contiguous here. So drop everything after "... all fields are required".